lib/lib.js

```diff
@@ -0,0 +1,201 @@
+var util = require('util');
+var os = require('os');
+
+var Transform = require('stream').Transform;
+
+var lstream = require('lstream');
+var request = require('request');
+var async = require('async');
+
+var github = new (require('github'))({
+  version: '3.0.0',
+  headers: {
+    'user-agent': 'git-apply-pr',
+  },
+});
+
+
+function authenticate(token) {
+  github.authenticate({
+    type: 'token',
+    token: token,
+  });
+}
+
+exports.getOutput = getOutput;
+function getOutput(type, config) {
+  var OUTPUT = {
+    'PR': util.format('#%d', config.PR),
+    'PR-URL': util.format('https://%s/%s/%s/pull/%d',
+      config.HOST, config.OWNER, config.REPO, config.PR),
+    'Reviewed-By': [],
+  };
+
+  return OUTPUT[type];
+}
+
+exports.getComments = getComments;
+function getComments(config, page, next) {
+
+  async.parallel({
+    issues: function getIssueComments(cb) {
+      github.issues.getComments({
+        user: config.OWNER,
+        repo: config.REPO,
+        number: config.PR,
+        per_page: 100,
+        page: page,
+      }, cb);
+    },
+    pr: function getPrComments(cb) {
+      github.pullRequests.getComments({
+        user: config.OWNER,
+        repo: config.REPO,
+        number: config.PR,
+        per_page: 100,
+        page: page,
+      }, cb);
+    }
+  },
+  function gotComments(err, comments) {
+    if (err) {
+      exitWithMsg(err);
+    }
+
+    var reviews = {};
+    comments = comments.issues.concat(comments.pr);
+    comments.forEach(function (comment) {
+      if (/lgtm/i.test(comment.body) ||
+          maybeTestForPlusone(config.ENABLE_PLUS_ONE, comment.body)) {
+        var val = reviews[comment.user.login] || 0;
+        reviews[comment.user.login] = val + 1;
+      }
+    });getOutput
+
+    if (comments.length < 100)
+      next(reviews, config);
+    else
+      getComments(page + 1, next);
+  });
+}
+
+function maybeTestForPlusone(ENABLE_PLUS_ONE, comment) {
+  if (!ENABLE_PLUS_ONE) {
+    return false;
+  }
+
+  return /\+1/.test(comment);
+}
+
+
+exports.resolveNames = resolveNames;
+function resolveNames(reviews, config) {
+  var queue = Object.keys(reviews);
+
+  var result = getOutput('Reviewed-By', config);
```

Comment:
I'm not sure the way `getOutput` is implemented is the right abstraction, since it returns a string most of the time, and an empty array object when `'Review-By'` is passed as its first parameter. In this case, we use `result.push` below, which means that `result` _has to_ be an array, and if `getOutput` changes its return value for `'Review-By'`, this code will break.

In other words, there's too much coupling between `getOutput('Reviewed-By', ...)` and its single, only usage. I would suggest splitting `getOutput` into several functions such as `formatReviewedByMetaData(userNames, prInfo, config)`, `formatPRMetaData(prInfo, config)` and `formatPRUrlMetaData(prInfo, config)`, which would all return a string.